src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1317,6 +1317,28 @@ bool AdaptDispatch::SetTopBottomScrollingMargins(const SHORT sTopMargin,
     return _DoSetTopBottomScrollingMargins(sTopMargin, sBottomMargin) && CursorPosition(1, 1);
 }
 
+// Routine Description:
+// - IND/NEL - Performs a line feed, possibly preceded by carriage return.
+//    Moves the cursor down one line, and possibly also to the leftmost column.
+// Arguments:
+// - lineFeedType - Specify whether a carriage return should be performed as well.
+// Return Value:
+// - True if handled successfully. False otherwise.
+bool AdaptDispatch::LineFeed(const DispatchTypes::LineFeedType lineFeedType)
+{
+    switch (lineFeedType)
+    {
+    case DispatchTypes::LineFeedType::DependsOnMode:
+        return !!_conApi->PrivateLineFeed(!!_conApi->PrivateGetLineFeedMode());
```

Comment:
Yeah, I'm not convinced the current solution is perfect, but I did consider other approaches too. I tried passing the _lineFeedType_ straight through to the `ConGetSet::PrivateLineFeed` method, but it didn't seem like we could easily share the type across those interfaces. The only `ConGetSet` method that does something similar is `SetCursorStyle`, and that's achieved with two different enums and then having to map from the one to the other, which felt icky to me.

Using `std::optional<bool>` would avoid that problem - assuming we use that everywhere in place of the current enum - but it feels a little less readable. I'm not totally opposed to that approach though.

That said, another argument I had against handling the logic in `ConGetSet::PrivateLineFeed` was I wanted to try and keep that interface as simple as possible, and make `AdaptDispatch` handle anything VT-specific. The idea being that `ConGetSet` could one day be the interface that differentiates between the conhost and Terminal implementations, and all the generic VT behaviour would be shared in `AdaptDispatch` (what I was suggesting in issue #3849). Under those conditions you want as little code as possible in the `ConGetSet` implementation, because everything on that side needs to be written twice.

Bottom line is I have a slight preference for the current approach. But if you'd like me to go another way, I don't might changing things.

Reply:
I'm willing to go with your existing slight preference.